lib/network.go

```diff
@@ -1914,6 +1915,14 @@ type MsgDeSoHeader struct {
 	// event that ASICs become powerful enough to have birthday problems in the future.
 	ExtraNonce uint64
 
+	// TransactionsConnectStatus is only used for Proof of Stake blocks, starting with
+	// MsgDeSoHeader version 2. For all earlier versions, this field will default to nil.
+	//
+	// The hash of the TxnConnectStatusByIndex field in MsgDeSoBlock. It is stored to ensure
+	// that the TxnConnectStatusByIndex is part of the header hash, which is signed by the
+	// proposer. The full index is stored in the block to offload space complexity.
+	TxnConnectStatusByIndexHash *BlockHash
```

Comment:
I get what you're trying to do here. But could we maybe integrate it into the existing TransactionMerkleRoot somehow? What I mean is maybe we can just change how we compute TransactionMerkleRoot when we cut over to PoS so that it includes a hash of the bitset you've added. It's kindof the point of the TransactionMerkleRoot root that it "rolls up" all the hashes you care about, which seems to include this one.

Maybe you can add bitsetHash as an optional argument to the function that computes the TransactionMerkleRoot. If you have a bitset, then it adds its hash to the merkle. Otherwise, it's the "old way" of doing the TransactionMerkleRoot.

Reply:
We discussed this with @tholonious and it felt like modifying the existing MerkleRoot utils to cater to hashing the new index will be tricky from the standpoint of backwards-compatibility and separation of concerns. 

A separate hash in the header adds just 32 bytes, which seems like an acceptable downside.

Reply:
Just adding here that we resolved this after the review in favor of sticking with the current field. We have a clear upgrade path for merging this into the merkle root when we introduce a new header version.